src/stream/src/executor/source/reader.rs

```diff
@@ -91,43 +80,44 @@ impl SourceReaderStream {
 
         Self {
             inner,
-            paused: None,
+            paused: false,
         }
     }
 
     /// Replace the source chunk reader with a new one for given `stream`. Used for split change.
     pub fn replace_source_chunk_reader(&mut self, reader: Box<SourceStreamReaderImpl>) {
-        if self.paused.is_some() {
-            panic!("should not replace source chunk reader when paused");
-        }
+        assert!(
+            !self.paused,
+            "should not replace source chunk reader when paused"
+        );
         *self.inner.get_mut().1 = Self::source_chunk_reader(reader).map(Either::Right).boxed();
     }
 
     /// Pause the source stream.
     pub fn pause_source(&mut self) {
-        if self.paused.is_some() {
-            panic!("already paused");
-        }
-        let source_chunk_reader =
-            std::mem::replace(self.inner.get_mut().1, Self::paused_source().boxed());
```

Comment:
Seems we can also avoid `Box<dyn>` for streams. 🤔

Reply:
I got `error[E0391]: cycle detected when computing type` when trying to use `type xx = impl Stream<...>`. It's really hard to make compiler happy. 🥵